### ISSUE TITLE
isAlphanum broken

### ISSUE THREAD

**Issue by tlbtlbtlb** [twolfson/node-jsmin2#6]
Changes to isAlphanum in commit 6c82b2756b01f878b05cc6472c67860bf831ad38 make it crash on most inputs. As a simple example:

```
> var jsmin = require('jsmin2');
> jsmin('{\n}\n')
TypeError: c.charCodeAt is not a function
    at isAlphanum (/usr/lib/node_modules/jsmin2/lib/jsmin.c.index.js:103:15)
    at jsmin (/usr/lib/node_modules/jsmin2/lib/jsmin.c.index.js:320:25)
    at main (/usr/lib/node_modules/jsmin2/lib/jsmin.c.index.js:373:9)
    at jsminFn (/usr/lib/node_modules/jsmin2/lib/jsmin.c.index.js:378:12)
    at jsmin (/usr/lib/node_modules/jsmin2/lib/jsmin.js:82:3)
    at repl:1:1
    at REPLServer.defaultEval (repl.js:252:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:417:12)
```

The thing being passed to isAlphanum which is not a string and so doesn't have the charCodeAt method is the number -1, the EOF marker. I think this is normal behavior of the original C code. isAlphanum should first check for -1 and return false.

**Comment by twolfson** [comment]
Would you be interested in submitting a PR to resolve this?

**Comment by tlbtlbtlb** [comment]
Done. https://github.com/twolfson/node-jsmin2/pull/7

**Comment by twolfson** [comment]
This has been resolved/patched/released in `1.2.1` via @tlbtlbtlb in #7
